### ISSUE TITLE
EditorPlugin forward methods are not called when editor selection is empty

### ISSUE THREAD

**Issue by ExquisiterEmil** [godotengine/godot#75050]
### Godot version

4.0.stable

### System information

Windows 10

### Issue description

I don't know if this is intended, but in case it is, I have a use case where this is obstructive. 

When you have an `EditorPlugin`, that is supposed to be active, regardless of the currently selected node, resource, file, etc., many built-in methods of `EditorPlugin` are not being called anymore. 

Example: you want to display data (let's say some custom performance stat)  next to the mouse, by calling `_forward_3d_draw_over_viewport()`. The text should be drawn, no matter what the inspector shows at the moment, so you just override the _handles method like this:
```python
func _handles(object):
    return true
```

 however, when there is *nothing* selected at all, the text is not being updated, because `_forward_3d_draw_over_viewport()` is not being called. 

The same applies to `_forward_3d_gui_input`, and probably the equivalent 2d methods and force draw, although I did not test that specifically.

In my specific use case, I have a plugin that places assets in the scene, and it should be available at all times, no matter the selection, because in parts of the plugin things are selected and deselected on purpose to allow the user to perform transformations in the 3D view and modifications in the inspector.

### Steps to reproduce

Write an editor plugin that overrides `_handles` such that it always returns true. Make use of some of the forwarding methods. Observe, how it is only called when *something* is selected.
E.g. with the minimum reproduction project, open test.tscn and observe by selecting and deselecting nodes, how text is only drawn over the 3D viewport, when the inspector shows that *something* has been selected (a file, a resource, a node, ...) 

### Minimal reproduction project

[editor_plugin_test_4.0.zip](https://github.com/godotengine/godot/files/11007382/editor_plugin_test_4.0.zip)

**Comment by KoBeWi** [comment]
This is intended. The handles/edit API is for context-based editors. They edit objects based on scene tree and editor contents and automatically stop editing when not needed anymore.

For plugins that have to be always active you can just use the regular `_process()`/`_input()` methods. EditorPlugin is a node, so you don't even need anything extra.
The only thing we are missing is some handle to editor overlays (the ones from `_forward_3d_draw_over_viewport()` and 2D). We could add `EditorInterface.get_2d_overlay()` (and 3D) to allow users hook up to the `draw` signal and manually draw stuff over the viewport. Right now you need to search the scene tree to find the overlay or use the gui base, which AFAIK draws over the whole editor.

**Comment by kleonc** [comment]
There's [`EditorPlugin.update_overlays`](https://docs.godotengine.org/en/4.0/classes/class_editorplugin.html#class-editorplugin-method-update-overlays) method which is supposed to be called to trigger redrawing the overlays.

**Comment by KoBeWi** [comment]
Yes, and it calls the forward draw methods in *active plugins*.

**Comment by TheDuckCow** [comment]
Sometime late, I've now also run into this. It's a bit odd, because it used to work in Godot 3.5 and took me awhile to understand during some code migration if that was just a bug or intended (until I came across this issue). I guess it was an oversight in 3.5 that happened to be rather convenient, since it got us notice of any 3d viewport interaction events without consequence.

The one unfortunate thing is that indeed we have to do a bit of hunting now - just switching to _input isn't quite as nice since, as mentioned, this checks inputs over the entire godot interface and not just the active 3d view we want to interact with (and even when the 3d view isn't open). I'm now trying to figure out how to isolate to just the 3D view so I'm not suddenly interrupting or responding to left button ups/downs in completely unrelated areas, but definitely more cumbersome than what I'd expect. 

In blender bpy scripting, we have the concept of an active "context" ([ref](https://docs.blender.org/api/current/bpy.context.html)) which itself always carries an "area" which itself contains class-specific member fields and so forth. I feel like such a concept. As a long time plugin developer in Blender, the ability to get context for actions can be quite useful, and seems relevant in such a scenario like this. Not sure if this idea is helpful or how it would look to implement, could be a more general purpose editor plugin method which is like _input but with additional args provided.

**Comment by KoBeWi** [comment]
`_input()` is only one of the options, there is also `_unandled_input()` etc. and, if you feel like doing more hacking, you can put a Control over the viewport and use `_gui_input()`.

**Comment by TheDuckCow** [comment]
Yeah I more meant that using using such methods respond to the entire interface. Your idea of inserting a control over the viewport editor node sounds like what I need to do, thanks for that tip.

Still, seems like a reasonable (even if slightly less common) option for plugins to respond to events within just a specific viewport, and "just work" with different layouts like quad view without plugin developers trying to hack in the solution to also be responsive to such toggles.

**Comment by IceflowRE** [comment]
There is
[`set_input_event_forwarding_always_enabled`](https://docs.godotengine.org/en/stable/classes/class_editorplugin.html#class-editorplugin-method-set-input-event-forwarding-always-enabled) and [`set_force_draw_over_forwarding_enabled`](https://docs.godotengine.org/en/stable/classes/class_editorplugin.html#class-editorplugin-method-set-force-draw-over-forwarding-enabled)

**Comment by Nodragem** [comment]
To have a `EditorInterface.get_3d_viewport_overlay()` function would be terribly useful when try to make plug-ins that draw stuffs on screen.
